src/tools/validator/custom-validations/valid-placeholders.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * This validation function loops recursively through all of the properties in the given input object
+ * and for all string values extracts and validates any existing placeholders.
+ *
+ * A valid placeholder is considered to look like this ``{{placeholderName}}``. So this validator looks
+ * for mistakes such as ``{placeHolderName}}`` or ``{{placeHolderName}}}``, for example.
+ *
+ * When such an occurrence is found an error message is generated indicating in which property
+ * the error occurred along with the offending placeholder.
+ *
+ */
+export const validPlaceholders = (
+  input: any,
+  parentName?: string
+): string[] | undefined => {
+  const parentSuffix = parentName ? `${parentName}.` : '';
+
+  const result = Object.entries(input)
+    .flatMap(([propertyName, value]: [string, any]) => {
+      if (typeof value === 'string') {
+        const result = validatePlaceHolders(value);
+        if (result.length) {
+          return result.map(
+            (placeholder: string) =>
+              `Invalid placeholder '${placeholder}' found in ${parentSuffix}${propertyName}`
+          );
+        }
+        return;
+      }
+
+      if (typeof value === 'object') {
+        return validPlaceholders(value, `${parentSuffix}${propertyName}`);
+      }
+    })
+    .filter(Boolean) as string[];
+
+  return result.length ? result : undefined;
+};
+
+function validatePlaceHolders(text: string): string[] {
+  const matches = [...(text.matchAll(/({[^}]+[}]+)/g) as any)];
+
+  return matches
+    .map((matchInfo: string[]) => {
+      const matched = matchInfo[0];
+      const prefix = matched.substr(0, 2);
+      const suffix = matched.substr(-2);
+      const middle = matched.substr(2, matched.length - 4);
+      if (
+        prefix !== '{{' ||
+        suffix !== '}}' ||
+        middle.indexOf('{') > -1 ||
+        middle.indexOf('}') > -1
+      ) {
+        return matched;
+      }
+      return undefined;
+    })
+    .filter(Boolean) as string[];
```

Comment:
```suggestion
    .filter(isDefined);
```

if you use isDefined from ts-is-present, then typescript can infer the correct return type and the code is more robust, because if you made a mistake in the map the compiler will not complain since you explicitly cast the return type.